diff --git commerce_paypal/modules/wps/commerce_paypal_wps.module commerce_paypal/modules/wps/commerce_paypal_wps.module
index 76d703f..3731334 100644
--- commerce_paypal/modules/wps/commerce_paypal_wps.module
+++ commerce_paypal/modules/wps/commerce_paypal_wps.module
@@ -13,7 +13,7 @@ function commerce_paypal_wps_commerce_payment_method_info() {
   $payment_methods = array();
 
   $icons = commerce_paypal_icons();
-  $display_title = t('!logo PayPal - pay securely without sharing your financial information', array('!logo' => $icons['paypal']));
+  $display_title = t('<p>When you click \'Continue to next step\' you will be taken to the PayPal payment gateway to securely process your payment.</p><p>If you don\'t have a PayPal account, don\'t worry, you can pay using your credit or debit card by clicking the \'Dont have a PayPal account?\' link.</p>', array('!logo' => $icons['paypal']));
   $display_title .= '<div class="commerce-paypal-icons"><span class="label">' . t('Includes:') . '</span>' . implode(' ', $icons) . '</div>';
 
   $payment_methods['paypal_wps'] = array(
